olm/operator-registry.md

```diff
@@ -0,0 +1,322 @@
+
+This database will be built by tooling that also needs to be created, but is not in scope of this enhancement.
+
+---------------------
+
+_**Serve Operator Database**_
+
+As an Operator developer, I want my Operator to be consumed and made available to clusters.
+
+- So that the metadata attached to my operator can be used to determine what is available to install on a cluster.
+
+Acceptance Criteria:
+I can create a registry database that hosts a history of my operator and other developer's operators. This database can be used by openshift clusters.
+
+---------------------
+
+_**Add Indexes to cluster**_
+
+As an OpenShift administrator, I want to have a way of adding an operator index to my cluster
+
+- So that I can make operators available to install onto my cluster
+
+Acceptance Criteria:
+When I add an operator index to my cluster, I am able to get information about the operators defined in that index.
+
+Internal detail note:
+Create an image type CatalogSource that, when added to a cluster with a reference to an index image, pulls the index manifest down and generates the operator metadata database for that operator.
+
+---------------------
+
+_**Drive Operator Installs and UX**_
+
+As an OpenShift administrator, I want a way to install operators made available to me from my Operator index.
+- So that I can view what operators are available to install
+- So that I can view metadata about that operator (description, author, etc) on the OpenShift console
+- So that I can select a given operator name, version, and channel to subscribe to
+
+Acceptance Criteria:
+
+When I install an operator index onto my cluster, I can use the data that is available on the cluster to drive operator installation and lifecycle.
+
+---------------------
+
+_**Polling Updates**_
+
+As an OpenShift administrator, I want a way to configure a poll timer to ensure my index of operators is up to date and references all of the latest operator bundles.
+
+- So that I can get updates over time
+- So that I can know the frequency that those updates will appear
+
+Acceptance Criteria:
+When I set the update frequency on an operator index, OLM will poll and ensure that if there is a new version of the index that those changes are pulled and propagated into the cluster.
+
+Internal details:
+Polling process should be reviewed by quay team to ensure we scale on quay (although these indexes should work on ANY container image registry)
+
+### Implementation Details
+
+The goal of this enhancement is to replace the app-registry implementation in a way that is usable by operator developers and operator author developers. To do this we will build an index image that is defined from a set of operator bundle images that are released independently and optimized in a way that can be productized by operator authors and pipeline developers.
```

Comment:
What's the difference between an operator developer and an operator author developer?

Reply:
Looks like it should be `operator authors and pipeline developers`

Reply:
Yep, this is a typo. I'll fix it in an update.